src/codegen/spirv/Module.zig

```diff
@@ -477,20 +581,93 @@ pub fn floatType(self: *Module, bits: u16) !IdRef {
+            .component_type = child_ty_id,
             .component_count = len,
         });
     }
     return entry.value_ptr.*;
 }
 
+pub fn arrayType(self: *Module, len_id: IdRef, child_ty_id: IdRef) !IdRef {
+    const entry = try self.cache.array_types.getOrPut(self.gpa, .{ child_ty_id, len_id });
+    if (!entry.found_existing) {
+        const result_id = self.allocId();
+        entry.value_ptr.* = result_id;
+        try self.sections.types_globals_constants.emit(self.gpa, .OpTypeArray, .{
+            .id_result = result_id,
+            .element_type = child_ty_id,
+            .length = len_id,
+        });
+    }
+    return entry.value_ptr.*;
+}
+
+pub fn functionType(self: *Module, return_ty_id: IdRef, param_type_ids: []const IdRef) !IdRef {
+    const entry = try self.cache.function_types.getOrPut(self.gpa, .{ return_ty_id, param_type_ids });
+    if (!entry.found_existing) {
+        const result_id = self.allocId();
+        entry.value_ptr.* = result_id;
+        try self.sections.types_globals_constants.emit(self.gpa, .OpTypeFunction, .{
+            .id_result = result_id,
+            .return_type = return_ty_id,
+            .id_ref_2 = param_type_ids,
+        });
+    }
+    return entry.value_ptr.*;
+}
+
+pub fn constBool(self: *Module, value: bool) !IdRef {
+    if (self.cache.bool_const[@intFromBool(value)]) |b| return b;
+
+    const result_ty_id = try self.boolType();
+    const result_id = self.allocId();
+    self.cache.bool_const[@intFromBool(value)] = result_id;
+
+    switch (value) {
+        inline else => |value_ct| try self.sections.types_globals_constants.emit(
+            self.gpa,
+            if (value_ct) .OpConstantTrue else .OpConstantFalse,
+            .{
+                .id_result_type = result_ty_id,
+                .id_result = result_id,
+            },
+        ),
+    }
+
+    return result_id;
+}
+
+pub fn constInt(self: *Module, signedness: std.builtin.Signedness, bits: u16, value: anytype) !IdRef {
+    const entry = try self.cache.int_consts.getOrPut(self.gpa, .{ .signedness = signedness, .bits = bits });
+    if (!entry.found_existing) {
+        const result_ty_id = try self.intType(signedness, bits);
+        const result_id = self.allocId();
+        entry.value_ptr.* = result_id;
+        switch (bits) {
+            0 => unreachable, // u0 is comptime
+            1...32 => try self.sections.types_globals_constants.emit(self.gpa, .OpConstant, .{
+                .id_result_type = result_ty_id,
+                .id_result = result_id,
+                .value = .{ .uint32 = @truncate(value) },
+            }),
+            33...64 => try self.sections.types_globals_constants.emit(self.gpa, .OpConstant, .{
+                .id_result_type = result_ty_id,
+                .id_result = result_id,
+                .value = .{ .uint64 = value },
+            }),
+            else => unreachable, // TODO: Large integer constants
+        }
+    }
+    return entry.value_ptr.*;
+}
```

Comment:
This doesn't seem correct to me, the int_consts map only caches the signedness and the amount of bits, not the value. If one does something like this
```
const a = spv.constInt(.unsigned, 16, 0);
const b = spv.constInt(.unsigned, 16, 1);
```
then `a` and `b` will return the same ID...